multitenancy/tenant-manager-api/src/main/java/io/apicurio/multitenant/api/RegistryTenantResource.java

```diff
@@ -0,0 +1,107 @@
+package io.apicurio.multitenant.api;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import io.apicurio.multitenant.datamodel.NewRegistryTenantRequest;
+import io.apicurio.multitenant.datamodel.RegistryTenant;
+import io.apicurio.multitenant.persistence.RegistryTenantRepository;
+import io.apicurio.multitenant.persistence.TenantNotFoundException;
+import io.apicurio.multitenant.persistence.dto.RegistryTenantDto;
+
+@Path("/v1/tenants")
+public class RegistryTenantResource {
+
+    @Inject
+    RegistryTenantRepository tenantsRepository;
+
+    @Inject
+    @ConfigProperty(name = "tenant.manager.auth.server.url")
+    Optional<String> defaultAuthServerUrl;
+
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<RegistryTenant> listTenants() {
+        return tenantsRepository.listAll()
+                .stream()
+                .map(RegistryTenantDto::toDatamodel)
+                .collect(Collectors.toList());
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createTenant(NewRegistryTenantRequest tenantRequest) {
+
+        required(tenantRequest.getOrganizationId(), "OrganizationId is mandatory");
+        required(tenantRequest.getClientId(), "ClientId is mandatory");
+
+        RegistryTenantDto tenant = new RegistryTenantDto();
+
+        tenant.setTenantId(UUID.randomUUID().toString());
+
+        tenant.setOrganizationId(tenantRequest.getOrganizationId());
+        tenant.setAuthClientId(tenantRequest.getClientId());
+
+        if (tenantRequest.getAuthServerUrl() == null && defaultAuthServerUrl.isPresent()) {
```

Comment:
I think we should discuss this. The way quarkus oidc works is that when a `null` configuration is returned then it falls back to the default auth config (defined in `application.properties`).

Reply:
ok that's good. So you suggest that if no `authServerUrl` is given to the tenant-manager it should leave the column empty so apicurio-registry will pick it's own default, no?

Reply:
Yep, that's it.